### ISSUE TITLE
wallet output table broken row

### ISSUE THREAD

**Issue by SWvheerden** [tari-project/tari#4670]
Somehow the utxo validation service does not update a row properly.
We need to verify that we catch all instances that happen like this and prevent it. 
```
[wallet::output_manager_service] WARN  Error completing UTXO Validation Protocol (Id: 14741583085246359368): InconsistentDataError("Output marked as mined, but mined_height or mined_in_block was empty")
```

We need to make sure updates are atomic

**Comment by SWvheerden** [comment]
Do we just take the "easy" way out and mark that row as unmined, clear the mined_height and mined_in_block?

we have been struggling to find how this happens, and when this happens utxo validation is broken forever.

**Comment by sdbondi** [comment]
May be fixed in #4686
